pytealutils/applications/application.py

```diff
@@ -0,0 +1,228 @@
+from inspect import getattr_static, signature
+from typing import Any, Callable, Final
+
+from pyteal import (
+    MAX_TEAL_VERSION,
+    ABIReturnSubroutine,
+    And,
+    App,
+    Approve,
+    Assert,
+    AssetHolding,
+    BitLen,
+    CallConfig,
+    Concat,
+    Expr,
+    Global,
+    Int,
+    MethodConfig,
+    Router,
+    Seq,
+    Subroutine,
+    SubroutineFnWrapper,
+    TealInputError,
+    TealType,
+    Txn,
+    TxnField,
+)
+
+from .application_schema import AccountState, ApplicationState
+
+HandlerFunc = Callable[..., Expr]
+_handler_config_attr: Final[str] = "__handler_config__"
+
+
+def get_handler_config(fn: HandlerFunc | ABIReturnSubroutine) -> dict[str, Any]:
+    handler_config = {}
+    if hasattr(fn, _handler_config_attr):
+        handler_config = getattr(fn, _handler_config_attr)
+
+    return handler_config
+
+
+def add_handler_config(fn: HandlerFunc | ABIReturnSubroutine, key: str, val: Any):
+    handler_config = get_handler_config(fn)
+    handler_config[key] = val
+    setattr(fn, _handler_config_attr, handler_config)
+
+
+class PrependExpr:
```

Comment:
Is this descriptor magic integral to the goals of the PR? Could similar outcomes be achieved using properties?

Reply:
Sounds like it probably is integral. I'll have a look later in the evening to get better understanding.